### PR TITLE
Add travis and codecov integration

### LINKED ISSUE #14: Add Travis and CodeCov integration
As a developer, I'd like to add Travis and CodeCov integration, so we can incrementally CI all the commits (including PRs) against Travis and also measure the test coverage.

**Note:**
Let's stick to what we have done in [SCDF](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/.travis.yml) for consistency.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add travis config yml
 - Enable travis service integration via https://travis-ci.org/profile
 - Update travis config with codecov execution
 - Add maven plugin for codecov

Resolves #14

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
The travis service is enabled for this project as you can see the travis CI build is starting with this PR. 

For codecov, I think we still need someone with the privilege to configure the webhook here: https://codecov.io/gh/spring-cloud/spring-cloud-stream-binder-aws-kinesis

**Comment by artembilan** [comment]
Well, the Travis has failed with the error:
```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:

[FATAL] Non-resolvable parent POM for org.springframework.cloud:spring-cloud-stream-binder-kinesis-parent:1.0.0.BUILD-SNAPSHOT: Could not find artifact org.springframework.cloud:spring-cloud-build:pom:1.3.3.BUILD-SNAPSHOT in sonatype-snapshots (https://oss.sonatype.org/content/repositories/snapshots/) and 'parent.relativePath' points at no local POM @ line 6, column 10

 @ 

[ERROR] The build could not read 1 project -> [Help 1]

[ERROR]   

[ERROR]   The project org.springframework.cloud:spring-cloud-stream-binder-kinesis-parent:1.0.0.BUILD-SNAPSHOT (/home/travis/build/spring-cloud/spring-cloud-stream-binder-aws-kinesis/pom.xml) has 1 error

[ERROR]     Non-resolvable parent POM for org.springframework.cloud:spring-cloud-stream-binder-kinesis-parent:1.0.0.BUILD-SNAPSHOT: Could not find artifact org.springframework.cloud:spring-cloud-build:pom:1.3.3.BUILD-SNAPSHOT in sonatype-snapshots (https://oss.sonatype.org/content/repositories/snapshots/) and 'parent.relativePath' points at no local POM @ line 6, column 10 -> [Help 2]
```
Any ideas?
Thanks

File: .settings.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<settings>
```
Comment (artembilan) [inline .settings.xml]: Why do we need this file?
If we'd have the reason in the commit message I wouldn't ask such a question.
Thanks

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+sudo: required
+cache:
+  directories:
+    - $HOME/.m2
+language: java
+jdk:
+  - oraclejdk8
+install: true
```
Comment (artembilan) [inline .travis.yml]: I think the problem in Travis report that we really don't do any `install` for artifacts in this project.
No?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+script:
+  - '[ "${TRAVIS_PULL_REQUEST}" != "false" ] || ./mvnw package -Pfull -Pspring -U -Dmaven.test.redirectTestOutputToFile=true'
+  - '[ "${TRAVIS_PULL_REQUEST}"  = "false" ] || ./mvnw package -Pfull -Pspring -U -Dmaven.test.redirectTestOutputToFile=true'
```
Comment (artembilan) [inline .travis.yml]: What is the logic behind this mutually exclusive commands?
Looks like they are the same, and one or another will be executed independently of the `TRAVIS_PULL_REQUEST` value.
I mean there is just enough to have:
```
script:
./mvnw package -Pfull -Pspring -U -Dmaven.test.redirectTestOutputToFile=true'
```

I won't mind if I miss something, but at least, please, explain how this should work.

Thanks

**Comment by ilayaperumalg** [comment]
@artembilan I think the problem is related to maven not pointing to the correct repo to download the artifacts. Added `-s .setting.xml` with settings.xml having spring repos.

**Comment by artembilan** [comment]
So, Travis works well now!

I need answers to my concerns to utilize them in the final commit message do not come back to this in the future projects and help readers to understand what we are doing here.

And also I can't merged this PR until the Codecov is configured properly.

Do you know someone who can help us?

Thanks

**Comment by ilayaperumalg** [comment]
@artembilan Codecov got triggered as part of the travis configuration as we can see the report there.

> I need answers to my concerns to utilize them in the final commit message do not come back to this in the future projects and help readers to understand what we are doing here.

Could you elaborate on this, please?

**Comment by artembilan** [comment]
> codecov/patch — Coverage not affected. 

> codecov/project — No report found to compare against 

Does it say anything to you?

Sorry, I don't understand that tool and how it should work and for what reason.
Since it fails right now, I'm afraid we have to fix something else.
I'd be glad to, but as I said: I don't know that tool.

That's why I'm seeking explanations and help.

The rest my concerns are on the line comment.
Don't you see them?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+cache:
+  directories:
+    - $HOME/.m2
+language: java
+jdk:
+  - oraclejdk8
+install: true
+script:
+  - '[ "${TRAVIS_PULL_REQUEST}" != "false" ] || ./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'
+  - '[ "${TRAVIS_PULL_REQUEST}"  = "false" ] || ./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'
```
Comment (artembilan) [inline .travis.yml]: Why do we need these two identical commands?
Looks like both one of them is performed independently of the `TRAVIS_PULL_REQUEST` value.

Or am I missing anything?

File: .settings.xml (hunk 1 above)
Comment (ilayaperumalg) [inline .settings.xml]: ;) This was added as part of later commits in this PR to fix pointing to spring repos.

File: .travis.yml (hunk 2 above)
Comment (ilayaperumalg) [inline .travis.yml]: we can remove this for now.

File: .travis.yml (hunk 4 above)
Comment (ilayaperumalg) [inline .travis.yml]: yeah, I copied this from SCDF where we had different flow PR and main commit. hence these lines. will remove on of them.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+cache:
+  directories:
+    - $HOME/.m2
+language: java
+jdk:
+  - oraclejdk8
+script:
+  - '[ "${TRAVIS_PULL_REQUEST}" != "false" ] || ./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'
```
Comment (artembilan) [inline .travis.yml]: @ilayaperumalg, 

please, take a look into the Travis report.
Is that really an expected behavior?
I would like to see some logs about building project, but there is no.

Maybe we don't need this `TRAVIS_PULL_REQUEST` condition at all?

Comment (ilayaperumalg) [inline .travis.yml]: yeah, I removed that condition now.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+cache:
+  directories:
+    - $HOME/.m2
+language: java
+jdk:
+  - oraclejdk8
+script:
+  - ./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'
```
Comment (artembilan) [inline .travis.yml]: Looks like something has not been fixed yet:
```
0.00s$ ./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'

/home/travis/.travis/job_stages: eval: line 57: unexpected EOF while looking for matching `''

/home/travis/.travis/job_stages: eval: line 58: syntax error: unexpected end of file

The command "./mvnw package -Pfull -s .settings.xml -U -Dmaven.test.redirectTestOutputToFile=true'" exited with 1.
```
I guess that `'` in the end if redundant and erroneous.

I would say can fix on merge, but we really can be sure only here on GH with an active PR to run Travis process.

Thanks

Comment (ilayaperumalg) [inline .travis.yml]: yeah, let's fix this on the PR itself.

**Comment by ilayaperumalg** [comment]
@artembilan The travis build looks ok now. I guess the codecov requires some base report to compare against and may be the next run against a different PR would generate a report.

**Pull request merged**